### PR TITLE
Edit filename

### PR DESCRIPTION
**Pull request by vzqzac**
### What does this implement/fix? ###

Implements the handling and logic to edit filenames in-place 

### Does this close any currently open issues? ###

Finishes #49 

### Explain your workflow ###

- Refactored JS strategy on page model
- Added logic to handle filename modifications on the backend
- Added logic to handle filename modifications on the frontend

Example
------

### CODE REVIEW

File: app/controllers/pages_controller.rb [hunk 1]
```diff
@@ -46,8 +58,8 @@ def set_page
     end
 
     # Never trust parameters from the scary internet, only allow the white list through.
-    def update_page_params
-      params.require(:page).permit(:content, :url, :file)
+    def permitted_update_page_params
```
Comment (MatheusRich) [inline app/controllers/pages_controller.rb]: I think there's no need to change this.

File: app/views/pages/show.html.erb [hunk 2]
```diff
@@ -18,7 +18,19 @@
...
+        <p class="filename"><%= @page.file_blob.filename %></p>
+
+        <%= content_tag :p, class: 'filename-edit-container hidden' do -%>
+          <%= text_field_tag 'filename', @page.file_blob.filename.base %>
+
+          <span id="file-extension">
+            <%= ".#{@page.file_blob.filename.extension}" %>
+          </span>
+
+          <button class="button" type="button" name="update-filename-button" onclick="updateFileName()">
```
Comment (MatheusRich) [inline app/views/pages/show.html.erb]: I was thinking that pressing "Enter" and/or clicking out of the text area could be better. What do you think?

Comment (VzqzAc) [reply app/views/pages/show.html.erb]: I thought of a more visual way on doing so, so the button came to my mind, but I can add an event listener to the text input too. On the other hand, I don't think updating on blur is very friendly on the UX

**Comment by VzqzAc** [comment]
Yes @MatheusRich I'll add tests! My bad on not do so since the beginning 😅

File: app/views/pages/show.html.erb (hunk 2 above)
Comment (MatheusRich) [reply app/views/pages/show.html.erb]: Add just the "Enter" option. I'll do some UX tests with friends and I'll change it if needed! Thanks.

**Comment by MatheusRich** [comment]
@VzqzAc No problem :smile:.

**Comment by VzqzAc** [comment]
@MatheusRich I updated this PR, will try to take some spare time tomorrow to add the necessary tests but I think it's the only remaining thing unless you feel something else needs to be addressed

**Comment by MatheusRich** [comment]
@VzqzAc Hey, are you still up for this?

**Comment by andre-filho** [comment]
@MatheusRich where can I find the coverage diff? Those checks above took me to nowhere

**Pull request closed without merging**